Create market.browse.mdx (#177)
comcode-org/hackmud_wiki: docs/scripting/trust scripts/market.browse.mdx

---
title: market.browse
---

## Description

market.browse shows upgrades currently for sale on the marketplace.

### Security Level

FULLSEC

## Syntax

### CLI

```
market.browse { tier: 1, seller: "trust", rarity: 0 }
```

### Script

```
#fs.market.browse({ tier: 1, seller: "trust", rarity: 0 })
```

### Parameters

At least one argument is required. If no arguments are input, market.browse will return a Failure object and detailed usage information. All arguments are optional. Any parameter that can take a number can also take MongoDB query conditionals, e.g. `chars: { $gte: 2000 }`.

#### tier

> Takes a number and specifies the tier of upgrades to be shown.

Reviewer: ```suggestion
The 'tier' argument takes a number and specifies the tier of upgrades to be shown.
```

#### seller

> Takes a username as a string and specifies the seller, for example to filter upgrades sold by TRUST.

Reviewer: ```suggestion
The 'seller' argument takes a username as a string and specifies the seller, for example to filter upgrades sold by TRUST.
```

#### listed_before

> Takes a Unix timestamp, e.g. `1490286018.35423`.

Reviewer: ```suggestion
The 'listed_before" argument takes a Unix timestamp, e.g. `1490286018.35423`.
```

#### listed_after

> Takes a Unix timestamp, e.g. `1490286018.35423`.

Reviewer: ```suggestion
The 'listed_after" argument takes a Unix timestamp, e.g. `1490286018.35423`.
```

#### rarity

> Takes a number ranging fromm 0 to 5 and specifies the rarity of upgrades to be shown.

Reviewer: ```suggestion
The 'rarity' argument takes a number ranging from 0 to 5 and specifies the rarity of upgrades to be shown.
```

#### cost

> Can take a GC string, a number, or MongoDB query conditionals, e.g. `cost: { $lte:500000 }`

Reviewer: ```suggestion
The 'cost' argument can take a GC string, a number, or MongoDB query conditionals, e.g. `cost: { $lte: 500000 }`
```

#### name

> Takes a string and specifies the name of upgrades to be shown.

Reviewer: ```suggestion
The 'name' argument takes a string and specifies the name of upgrades to be shown.
```

#### type

> Takes a string and filters by upgrade type, e.g. `type: "lock"`.

Reviewer: ```suggestion
The 'type' argument takes a string and filters by upgrade type, e.g. `type: "lock"`.
```

#### class

> Takes a string and filters by upgrade class, e.g. `class: "architect"`

Reviewer: ```suggestion
The 'class' argument takes a string and filters by upgrade class, e.g. `class: "architect"`
```

#### chars

> Takes a number or MongoDB query conditional, `chars: { $gte: 2000 }`

Reviewer: ```suggestion
The 'chars' argument takes a number or MongoDB query conditional, `chars: { $gte: 2000 }`
```

#### i

> Takes a market token as a string and returns detailed information about a market upgrade.

Reviewer: ```suggestion
The 'i' argument takes a market token as a string and returns detailed information about a market upgrade.
```
